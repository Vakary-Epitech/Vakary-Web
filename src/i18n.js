import { createI18n } from 'vue-i18n'
import fr from './locales/fr.json'
import en from './locales/en.json'
import es from './locales/es.json'
import de from './locales/de.json'

export default createI18n({
  legacy: false,
  locale: 'fr',
  fallbackLocale: 'fr',
  globalInjection: true,
  messages: {
    en: en,
    fr: fr,
    es: es,
    de: de
  }
});