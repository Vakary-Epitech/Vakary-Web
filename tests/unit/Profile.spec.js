import { mount  } from '@vue/test-utils'
import Profile from '@/components/UI/profileModal.vue'
import i18n from '../../src/i18n.js'
import { createRouter, createWebHistory } from 'vue-router'

describe('Profile', () => {
  let wrapper
  const mockStore = {
    dispatch: jest.fn().mockResolvedValue(),
    state: {
      userStore: {
        userIsLoggedIn: true,
        user: {
          id: '123',
          name: 'John Doe',
          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed et commodo justo.',
          email: 'john.doe@example.com',
          image: 'https://via.placeholder.com/150'
        }
      }
    },
  };
  const router = createRouter({
    history: createWebHistory(),
    routes: [
      {
        path: '/',
        name: 'Profile',
        component: Profile,
      },
      {
        path: '/landingPage',
        name: 'landingPage'
      }
    ]
  })
  
  beforeEach(() => {
    wrapper = mount (Profile, {
      global: {
        plugins: [i18n, router],
        mocks: {
            $router: router,
            $store: mockStore,
        }
      }
    })
  });
  
  afterEach(() => {
    if (wrapper) {
      wrapper.unmount();
    }
  });

  it('renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  }
  );

  
})
