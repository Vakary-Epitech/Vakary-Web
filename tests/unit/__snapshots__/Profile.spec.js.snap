// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders the component 1`] = `
<div class="ProfilCardDesign">
  <div class="basicContainer">
    <div class="container">
      <div class="row">
        <div><img class="profile-picture" alt="profile-picture"></div>
        <div class="text-center">
          <h1 class="overflow" style="max-width: 200px;">test</h1>
        </div>
        <div class="col-12 overflow"><span>Description</span><br>
          <p class="description">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed et commodo justo.</p>
        </div>
      </div>
      <!--v-if-->
      <div class="row">
        <div class="row">
          <div class="col-5 offset-1"><span>Commentaires</span><br>
            <p></p>
          </div>
          <div class="col-3 offset-2"><span>J'aime</span><br>
            <p></p>
          </div>
        </div>
        <!--v-if-->
        <!--v-if-->
        <div class="col-12"><span>Km total</span><br>
          <p></p>
        </div>
      </div>
      <div class="col-12 my-2"><button>Editer le profil</button></div>
      <div class="col-12 my-2"><button>DÃ©connexion</button></div>
      <div class="col-12 mt-2"><button>Supprimer le compte</button></div>
    </div>
  </div>
</div>
`;
